api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -117,37 +77,97 @@ public Condition getCondition(Integer conditionId) {
 	}
 
 	/**
-     * @see ConditionService#getAllConditions(Patient)
+	 * @see ConditionService#getAllConditions(Patient)
 	 */
 	@Override
 	@Transactional(readOnly = true)
 	public List<Condition> getAllConditions(Patient patient) {
 		return conditionDAO.getAllConditions(patient);
 	}
 
+	/**
+	 * @see ConditionService#getConditionsByEncounter(Encounter)
+	 */
+	@Override
+	public List<Condition> getConditionsByEncounter(Encounter encounter) throws APIException {
+		return conditionDAO.getConditionsByEncounter(encounter);
+	}
+
+	/**
+	 * Saves a condition
+	 *
+	 * @param condition - the condition to be saved
+	 * @return the saved condition
+	 */
+	@Override
+	public Condition saveCondition(Condition condition) throws APIException {
+		Condition existingCondition = Context.getConditionService().getConditionByUuid(condition.getUuid());
+		// If there is no existing condition, then we are creating a condition
+		if (existingCondition == null) {
+			return conditionDAO.saveCondition(condition);
+		}
+
```

Comment:
Why do we need to support the void/unvoided functionality here?

Reply:
Essentially this is here because CoreApps still talks to the platform through the EMR API module's REST interface which only calls the `saveCondition()` method regardless of if the condition is updated, voided, etc. So supporting this here was the best way to keep the changes pretty independent of one another.